js/tests/unit/tab.spec.js

```diff
@@ -277,55 +301,6 @@ describe('Tab', () => {
 
       firstTab.show()
     })
-
```

Comment:
this test seems to test something that was never documented as a kosher technique/structure? it's invalid markup (having buttons inside of links) and should never be used, and i don't think we show this in any docs/examples (checked old v2 and v3 docs too, and couldn't see any example of this). feels safe to remove?

Reply:
This one?
**_should not fire hidden when hide is prevented_**

Reply:
yeah but not using close buttons this way. maybe programmatically removed ones?

Reply:
i've put the code for this and the dropdown tabs back for now...will need some more careful work here (to reformulate this one into a programmatic handling of tabs disappearing/being removed, and the other one a "breaking" change that should happen when we remove the actual logic in tab.js that handles this as well)